### PR TITLE
Donor chart, more share options, and deleting inactive rewards

### PR DESCRIPTION
**Pull request by benbacardi**

### CODE REVIEW

File: Shared/Views/DonorChart.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    let donations: [TiltifyDonorsForCampaignDonation]
+    let total: TiltifyAmount
+    
+    @State private var maxValue: Double = 0
+    @State private var minValue: Double = 0
+    @State private var showDonationValues: Bool = false
+    @State private var chartValues: [DonorChartValue] = []
+    
+    var body: some View {
+        if #available(iOS 16.0, *) {
```
Comment (dwrss) [inline Shared/Views/DonorChart.swift:28]: I'd have just annotated the entire struct, tbh

File: Shared/Views/DonorChart.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            .onChange(of: showDonationValues) { newValue in
+                calculateChartValues()
+            }
+        } else {
+            Text("Sorry!")
+        }
+    }
+    
+    func calculateChartValues() {
+        let reversedDonations = donations.filter {
```
Comment (dwrss) [inline Shared/Views/DonorChart.swift:63]: Are they reversed?

**Review by dwrss — APPROVED** [review]

File: Shared/Views/DonorChart.swift (hunk 1 above)
Comment (benbacardi) [reply Shared/Views/DonorChart.swift:28]: Yeah that's fair 😅

File: Shared/Views/DonorChart.swift (hunk 2 above)
Comment (benbacardi) [reply Shared/Views/DonorChart.swift:63]: Yeah, in the API and the list they're "most recent first" but for the graph we need "oldest first"

Comment (dwrss) [reply Shared/Views/DonorChart.swift:63]: But the filter operation doesn't reverse them?

**Pull request merged**